routes/auth-routes.js

```diff
@@ -0,0 +1,115 @@
+require("dotenv").config() 
+const express = require("express")
+const bcrypt = require("bcrypt")
+const jwt = require('jsonwebtoken')
+const UserModel = require("../models/user")
+const router = express.Router()
+
+router.use(express.json())
+
+
+// Register new user
+router.post("/register", async (req, res) => {
+   try {
+    // Get user input
+    const { name, email, password, role } = req.body
```

Comment:
User should not be able to pick their role. 
For shoppers they can sign up so role should default be shopper. 
For administrators they are entered into the database during production.
For staff they are created by admin.

Reply:
I only added role in there to see how to do the authorisation, ill remove it.